kv/local_test_cluster.go

```diff
@@ -59,6 +59,14 @@ func (rls *retryableLocalSender) Send(call client.Call) {
 		Tag:         fmt.Sprintf("routing %s locally", call.Method()),
 		MaxAttempts: 2,
 	}
+	// In local tests, the RPCs are not actually sent over the wire. We
+	// need to copy the Txn in order to avoid unexpected sharing between
+	// TxnCoordSender and client.Txn.
+	if header := call.Args.Header(); header.Txn != nil {
+		txn := &proto.Transaction{}
+		*txn = *header.Txn
```

Comment:
The shallow copy is enough to silence the race detector. Even easier than using marshal/unmarshal is to use `Clone`. Changed.